synapse/rest/media/v1/media_repository.py

```diff
@@ -157,11 +159,31 @@ def _download_remote_file(self, server_name, media_id):
                 try:
                     length, headers = yield self.client.get_file(
                         server_name, request_path, output_stream=f,
-                        max_size=self.max_upload_size,
+                        max_size=self.max_upload_size, args={
+                            # tell the remote server to 404 if it doesn't
+                            # recognise the server_name, to make sure we don't
+                            # end up with a routing loop.
+                            "allow_remote": "false",
+                        }
                     )
+                except twisted.internet.error.DNSLookupError as e:
+                    logger.warn("HTTP error fetching remote media %s/%s: %r",
+                                server_name, media_id, e)
+                    raise NotFoundError()
```

Comment:
well, it *could* be a transient error. My thinking was that it was more likely a permanent error: an invalid hostname, for example - in which case a 4xx is a better response, and NotFoundError seemed the closest fit.

Annoyingly I don't think `twisted.internet.endpoints.HostnameEndpoint` gives us a way of distinguishing the two cases. We're already wrapping `HostnameEndpoint` with our own implementation, so we could conceivably improve on this - but I thought it better to go with the common case.